### PR TITLE
🐛 Reject merge requests with an empty document list

### PR DESCRIPTION
**Pull request by tomhoule**
This makes the whole process fail earlier and gives better feedback.

A lot of the diff is from switching to the stable version of rustfmt.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,10 @@
 
 ## Unreleased
 
-## [0.3.1] - 2018-08-27
```
Comment (peterfication) [inline CHANGELOG.md]: Why did you remove that?

Comment (tomhoule) [reply CHANGELOG.md]: It hasn't been tagged nor released.

Comment (tomhoule) [reply CHANGELOG.md]: I think Fausto forgot.

Comment (brainlessdeveloper) [reply CHANGELOG.md]: Oops

File: tests/merge/end_to_end.rs [hunk 2]
```diff
@@ -157,3 +153,66 @@ pub fn test_end_to_end() {
     let res = core.run(tests).expect("tests failed");
     assert_eq!(res.status(), hyper::StatusCode::Ok);
 }
+
+pub fn test_rejection() {
```
Comment (brainlessdeveloper) [inline tests/merge/end_to_end.rs]: You should coach people on how to handle rejection.

Comment (tomhoule) [reply tests/merge/end_to_end.rs]: they can handle it with much fewer lines with a framework (I'll show myself to the door)

Comment (brainlessdeveloper) [reply tests/merge/end_to_end.rs]: D:
